adr/0017-zarf-bundle.md

```diff
@@ -0,0 +1,119 @@
+# 17. Bundles
+
+Date: 2023-06-13
+
+## Status
+
+Pending
+
+## Context
+
+Orchestrating capabilities from multiple Zarf packages into meta-packages is a current weak point for Zarf. The core of Zarf was built around components as capabilities, but as Zarf packages have scaled, there has been a need to create a new boundary to manage these capabilities efficiently.
+
+Currently there is no official way to enable the deployment, publishing, pulling, and creation of multiple Zarf packages together, and due to this some in the community have resorted to patterns such as:
+
+```yaml
+- name: init
+  required: true
+  files:
+    - source: zarf-init-amd64-v0.27.0.tar.zst
+      target: zarf-init-amd64-v0.27.0.tar.zst
+  actions:
+    onDeploy:
+      after:
+        - cmd: zarf package deploy zarf-init-amd64-v0.27.0.tar.zst --components git-server --confirm -l warn
+```
+
+While this _does_ fulfill the need to deploy two packages in one command, it does so in such a way that is verbose within the `zarf.yaml`, brittle across Zarf versions, inefficient within the package structure (it doesn't share layers), and is difficult to use `variables` with.
+
+### Proposed Solutions
+
+There are currently three proposed solutions to this, each with their own pros and cons:
+
+#### Zarf Bundle
+
+Uses OCI and a separate Zarf schema / declarative YAML definition to pull packages into a single artifact and orchestrate them together.
+
+Pros:
+
+- maintains efficient OCI layering / deduping of shared package resources
+- allows flexibility in defining what `zarf bundle` (or a separate command) would look like as its own command without polluting `zarf package`
+
+Cons:
+
+- variables set within packages with `setVariables` may be difficult to share across packages
+- package sources in bundles would be OCI _only_, and would not support local packages.
```

Comment:
This read like we are technically incapable of using local packages when creating bundles. That is not the case.


I get that we might have a preference for not using local packages but we should be explicit and list out our reasons for not wanting to rather than just stating that bundles won't support local packages.

Reply:
<3